src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala

```diff
@@ -343,17 +344,19 @@ class PeerManagerActor(
     prunedConnectedPeers
   }
 
-  private def getPeers(peers: Set[Peer]): Future[Peers] = {
-    implicit val timeout: Timeout = Timeout(2.seconds)
+  private def getPeers(peers: Set[Peer]): Task[Peers] = {
+    Task
+      .parSequence(peers.map(getPeerStatus))
+      .map(_.collect { case Success(v) => v }.toMap)
+      .map(Peers.apply)
+  }
 
-    Future
-      .traverse(peers) { peer =>
-        (peer.ref ? PeerActor.GetStatus)
-          .mapTo[PeerActor.StatusResponse]
-          .map { sr => Success((peer, sr.status)) }
-          .recover { case ex => Failure(ex) }
-      }
-      .map(r => Peers.apply(r.collect { case Success(v) => v }.toMap))
+  private def getPeerStatus(peer: Peer): Task[Try[(Peer, PeerActor.Status)]] = {
```

Comment:
Maybe we can replace Try with Either. 
What do you think?

Reply:
Well actually the exception in this code is never used so I'll just make it an `Option`

Reply:
sure, I like it